### ISSUE TITLE
About using dns hijack over ZeroTier

### ISSUE THREAD

**Issue by fanvinga** [zerotier/ZeroTierOne#980]
I usually run a device on my LAN and it provides a transparent gateway service.To use this service, I usually need to make the following settings on other devices in the LAN.What I want to do now is to use zerotier to build the network and achieve the same results as usual.
```
IP:192.168.1.3(Automatic allocation from dhcp)
Mask:255.255.255.0
Gateway:192.168.1.2(The IP of the service mentioned above)
DNS:198.18.0.2(The service use this address to hijack dns query and handle all traffic from remote machine) 
```
I have tried to add and configure both devices in the zerotier center and added the following routing table.
```
0.0.0.0/0 via 192.168.195.253 (Gateway device IP from zerotier private network)
```
Then i turn on `allowDefaultRoute` and add custom dns server `198.18.0.2`.But I can't find any devices on the gateway device that are trying to link to the service.

**Comment by fanvinga** [comment]
As far as the support of zerotier is concerned, may I achieve my goal?

**Comment by adamierymenko** [comment]
I'm not sure I understand... what is 192.168.195.253? I don't see it listed.

**Comment by fanvinga** [comment]
> I'm not sure I understand... what is 192.168.195.253? I don't see it listed.

192.168.195.0/24 is the Private-Network Range that Zerotier is using.
192.168.195.253 is the ip that the gateway-service-provider device using in zerotier's network.

Sorry for my poor English ability :-(

**Comment by adamierymenko** [comment]
On the host does it even attempt to route through this IP? E.g. if you trace route do you see it?

Also did you try the latest 1.4.0 release?

**Comment by fanvinga** [comment]
> On the host does it even attempt to route through this IP? E.g. if you trace route do you see it?
As the route table print i see that traffic has been redirect to Gateway IP.

> Also did you try the latest 1.4.0 release?
Just noticed that Zerotier release version 1.4.0.1,according to the changelog, I think the fix mentioned in 1.4.0.1 may be related to the problem I encountered.The version I've tried is 1.2.12.I'll try 1.4.0.1 ASAP :-D

Thank you for your patience anyway. :-D
